### PR TITLE
fix(pages): only owner and admin can change page access

### PR DESCRIPTION
**Pull request by jeabakker**

### CODE REVIEW

File: mod/pages/views/default/forms/pages/edit.php [hunk 1]
```diff
@@ -13,6 +13,11 @@
 $entity = elgg_extract('entity', $vars);
 $parent_guid = elgg_extract('parent_guid', $vars);
 
+$can_change_access = true;
+if ($entity instanceof ElggPage && $user instanceof ElggUser) {
+	$can_change_access = ($user->isAdmin() || $entity->owner_guid === $user->guid);
```
Comment (jdalsem) [inline mod/pages/views/default/forms/pages/edit.php]: why not $entity->getOwnerEntity()->canEdit()?

**Review by jdalsem — APPROVED** [review]

**Review by jdalsem — CHANGES_REQUESTED** [review]
see comment

File: mod/pages/views/default/forms/pages/edit.php (hunk 1 above)
Comment (jeabakker) [reply mod/pages/views/default/forms/pages/edit.php]: this is the logic used in the action, so keeping it the same

Comment (jdalsem) [reply mod/pages/views/default/forms/pages/edit.php]: you can also change it in the action...

**Pull request merged**